scrapbook/tests/test_notebooks.py

```diff
@@ -51,10 +50,29 @@ def test_bad_path():
 
 
 def test_bad_ext():
-    with pytest.raises(ValueError):
+    with pytest.raises(Warning):
         Notebook("not/a/valid/extension.py")
 
 
+@mock.patch("papermill.iorw.papermill_io.read")
+def test_good_ext_for_url(mock_read):
+    params = "?sig=some-unique-secret-token"
+    url = "abs://mystorage.blob.core.windows.net/my-actual-notebook.ipynb" + params
+    try:
+        Notebook(url)
+    except TypeError:
+        pass  # this means no valid json is returned by the reader
```

Comment:
Sorry to nitpick, but instead we should do `mock_read.return_value = <some notebook string>` above and then you won't raise any exception here.

Reply:
thanks, that makes more sense. Props for the very nice devops setup too!